### PR TITLE
add editorial + cpp code to angry cows

### PR DESCRIPTION
**Pull request by ryanchou-dev**
washed

---

_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by ryanchou-dev** [comment]
#4165

File: solutions/gold/usaco-597.mdx [hunk 1]
```diff
@@ -17,6 +35,141 @@ The second and third solutions in the analysis fail on the following test case:
...
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+using ll = long long;
+#define all(x) begin(x), end(x)
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-597.mdx:59]: please no

Comment (envyaims) [reply solutions/gold/usaco-597.mdx:59]: seethe and cope

File: solutions/gold/usaco-597.mdx [hunk 2]
```diff
@@ -17,6 +35,132 @@ The second and third solutions in the analysis fail on the following test case:
...
+	/**
+	 * @returns whether it's possible to explode
+	 * the rest of the haybales before/after idx
+	 *
+	 * @param pos position of our cow
+	 * @param idx furthest haybale we hit with our radius
+	 * @param r current radius at our position
+	 * @param dir whether we're checking positions to the right/left
+	 */
+	function<bool(int, int, int, bool)> push = [&](int pos, int idx, int r,
```
Comment (envyaims) [inline solutions/gold/usaco-597.mdx:89]: why not just make this global lmao? it's not like there are multiple test cases

File: solutions/gold/usaco-597.mdx (hunk 1 above)
Comment (ryanchou-dev) [reply solutions/gold/usaco-597.mdx:59]: 👀 do you want me to remove it?

**Review by envyaims — APPROVED** [review]
weird sol but ok

File: solutions/gold/usaco-597.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/gold/usaco-597.mdx:59]: please

File: solutions/gold/usaco-597.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/gold/usaco-597.mdx:89]: avoid global variables
and tbh it doesn't take away too much from readability

**Comment by SansPapyrus683** [comment]
could you test the code again? pushed some changes

**Comment by ryanchou-dev** [comment]
image

**Pull request merged**